src/SnapKeyring.test.ts

```diff
@@ -1,5 +1,9 @@
 import { TransactionFactory } from '@ethereumjs/tx';
-import { KeyringAccount } from '@metamask/keyring-api';
+import {
+  EthMethod,
+  EthAccountType,
+  InternalAccount,
+} from '@metamask/keyring-api';
```

Comment:
The tests regarding `serialize` and `deserialize` are not working currently because of the data type change.

Reply:
@danroc @montelaidev regarding our conversation earlier today, when should the `metadata` attribute be added to the accounts object to comply with the `InternalAccount`? And what's the source of data for the fields `snap` and `name`?

Reply:
`snap` metadata is filled by the `SnapKeyring` (this repo), the `name` should be filled by the `AccountsController`.